pipelines/matomo/helpers/data_processing.py

```diff
@@ -0,0 +1,84 @@
+"""Helpers for processing data from API calls."""
+
+from typing import Iterator, List
+import dlt
+from dlt.common import logger
+from dlt.common.pendulum import pendulum
+from dlt.common.typing import DictStrAny, TDataItem
+
+FIRST_DAY_OF_MILLENNIUM = "2000-01-01"
+FIRST_DAY_OF_MILLENNIUM_TIMESTAMP = 946684800.0
+
+
+def process_report(report: Iterator[TDataItem]) -> Iterator[TDataItem]:
+    """
+    Helper, loops through multiple formats of method_data and processes them into dlt resources
+    :param report: Response from Matomo API containing data for a single method.
+    :returns: generator of dicts.
+    """
+    if isinstance(report, dict):
```

Comment:
decomposing reports into single lines yielded separately will slow your pipeline substantially. the best way is to add elemenets to a list and then return a list (not yield). there's really no need to yield here. I know it looks cool but it is rather impractical